pkg/controllers/provisioningv2/rke2/machineprovision/controller.go

```diff
@@ -248,6 +380,7 @@ func (h *handler) OnRemove(_ string, obj runtime.Object) (runtime.Object, error)
 	}
 
 	// ErrSkip will not remove finalizer but treat this as currently reconciled
+	logrus.Infof("MachineProvision returning ErrSkip")
```

Comment:
I would advocate for removing this log message or adding more information to it (like machine name).